models/things.py

```diff
@@ -241,6 +241,84 @@ def get_by_mqtt_topic(db, topic: T.List[str]):
         return Thing.get_by_type_and_device_id(db, node_type, device_id, vnode_id), None
 
 
+class ShellyPower(Thing):
+    __mapper_args__ = {
+        'polymorphic_identity': 'shelly_power'
+    }
+
+    def get_data_type(self):
+        return DataType.Float
+
+    def get_state_topic(self):
+        return "shellies/{device_id}/emeter/{vnode_id}/power".format(type=self.type, device_id=self.device_id,
+                                                              vnode_id=self.vnode_id)
+
+    def get_action_topic(self):
+        return None
+
+    def process_status(self, db, state, data):
+        LastSeen.update_last_seen(db, self.device_id)
+        state = f"unknown,{state}"
+        print(f"State of {self.name} of type {self.type} is {state}")
+        return super().process_status(db, state, data)
+
+    @classmethod
+    def display_name(cls):
+        return 'Shelly Power'
+
+    @staticmethod
+    def get_mqtt_subscriptions():
+        return 'shellies/+/emeter/+/power',
```

Comment:
We could support Shelly Plug here in theory by returning a list here that also includes `"shellies/+/relay/+/power"`

Reply:
Tried this. The issue is that the relay state topic is
`shellies/<model>-<deviceid>/relay/0` but power and energy are
`shellies/<model>-<deviceid>/relay/0/power` and
`shellies/<model>-<deviceid>/relay/0/energy` respectivly. 
This is unfortunately not covered by our model.